app/controllers/companies_controller.rb

```diff
@@ -38,7 +38,7 @@ def index
 
     @companies = @all_companies.page(params[:page]).per_page(items_per_page)
 
-    render partial: 'companies_list' if request.xhr?
+    render partial: 'companies_list', locals: { companies: @companies } if request.xhr?
```

Comment:
I think that it's better to use more rails version with respond_to https://apidock.com/rails/ActionController/MimeResponds/InstanceMethods/respond_to
you can use shorter version of render
render 'companies_list', companies: @companies
http://api.rubyonrails.org/classes/ActionView/PartialRenderer.html#class-ActionView::PartialRenderer-label-Rendering+the+default+case

Reply:
Yes - I kept it this way because your (coming) PR will change this with the introduction of of `js.erb` file.  So, I kept this is the meantime.  We'll change it with your PR.